python/torch_mlir/dialects/torch/importer/jit_ir/csrc/torch_to_mlir_utils.cpp

```diff
@@ -403,8 +405,27 @@ MlirAttribute torch_mlir::importAttribute(MlirLocation loc,
 
 MlirLocation torch_mlir::getMlirLocationFromNode(MlirContext context,
                                                  torch::jit::Node *node) {
-  auto flc = node->sourceRange().file_line_col();
-  if (flc) {
+  if (node->hasAttribute(c10::Symbol::attr("source_files"))) {
+    const auto& source_files = node->ss(c10::Symbol::attr("source_files"));
+    const auto& line_numbers = node->is(c10::Symbol::attr("line_numbers"));
+    const auto& functions = node->ss(c10::Symbol::attr("functions"));
+
+    MlirLocation loc = mlirLocationUnknownGet(context);
+    for (const auto i : c10::irange(source_files.size())) {
+      MlirLocation new_loc = mlirLocationNameGet(
+        context,
+        toMlirStringRef(functions[i]),
+        mlirLocationFileLineColGet(
+          context,
+          toMlirStringRef(source_files[i]),
+          line_numbers[i],
+          0 /* column is not available */
+        )
+      );
+      loc = (i == 0 ? new_loc : mlirLocationCallSiteGet(new_loc, loc));
```

Comment:
Can you explain what is happening here? (Adding a comment would be useful). Is the idea that multiple `source_files` represents a call stack, so you have to chain the call site locations?

Reply:
Yeah, I'll add some comment here.

> Is the idea that multiple source_files represents a call stack, so you have to chain the call site locations?

Right, LTC gives us full stack trace per operation, so we need to chain it to get a sequence of `callsite` like that:
```
loc(callsite("forward"("pytorch/torch/nn/modules/linear.py":114:0) at callsite("_call_impl"("pytorch/torch/nn/modules/module.py":1488:0) at callsite("forward"("./torch-mlir/examples/ltc_backend_mnist.py":38:0) at callsite("_call_impl"("pytorch/torch/nn/modules/module.py":1488:0) at callsite("main"("./torch-mlir/examples/ltc_backend_mnist.py":56:0) at "<module>"("./torch-mlir/examples/ltc_backend_mnist.py":105:0)))))))
```